include/seqan3/alignment/matrix/detail/trace_iterator_banded.hpp

```diff
@@ -0,0 +1,107 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::trace_iterator_banded.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/matrix/detail/trace_iterator_base.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A trace iterator for banded trace matrices.
+ * \ingroup alignment_matrix
+ *
+ * \tparam matrix_iter_t The wrapped matrix iterator; must model seqan3::detail::two_dimensional_matrix_iterator and
+ *                       the iterator's value type must be the same as seqan3::detail::trace_directions, i.e.
+ *                       `std::same_as<value_type_t<matrix_iter_t>, trace_directions>` must evaluate to `true`.
+ *
+ * \details
+ *
+ * This iterator follows a given trace in a banded trace matrix.
+ */
+template <two_dimensional_matrix_iterator matrix_iter_t>
+class trace_iterator_banded : public trace_iterator_base<trace_iterator_banded<matrix_iter_t>, matrix_iter_t>
+{
+private:
+
```

Comment:
We don't usually have empty lines after private/public, just before, but that's :nail_care: